atomics/CMakeLists.txt

```diff
@@ -57,6 +60,9 @@ target_include_directories (desul_atomics PUBLIC
   $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
   $<INSTALL_INTERFACE:include>
   )
+if (BUILD_SHARED_LIBS)
+  target_compile_definitions(desul_atomics PRIVATE DESUL_EXPORT_SYMBOLS)
```

Comment:
Why do we need both `DESUL_BUILD_SHARED_LIBS` and `DESUL_EXPORT_SYMBOLS`?
Why do we put the first one in the config file and the second as a compile definition?

Reply:
We only need to export/import anything when building a shared library otherwise we just create overhead.
We need to export when building the library and import when using it. Thus, we can't control the behavior using a config file but must rather use build flags (that aren't exported).

Reply:
What overhead? Is there a downside to just always applying export or import when in win32?